src/flb_output.c

```diff
@@ -601,15 +601,15 @@ int flb_output_set_property(struct flb_output_instance *ins,
             if (strcasecmp(tmp, "false") == 0 ||
                 strcasecmp(tmp, "off") == 0) {
                 /* No limits for retries */
-                ins->retry_limit = -1;
+                ins->retry_limit = FLB_OUT_RETRY_UNLIMITED;
             }
             else {
                 ins->retry_limit = atoi(tmp);
             }
             flb_sds_destroy(tmp);
         }
         else {
-            ins->retry_limit = 0;
+            ins->retry_limit = FLB_OUT_RETRY_NONE;
         }
```

Comment:
What user input leads to this else case being run? I'm trying to follow the logic and I don't get it..

Reply:
We can set via command line. `-p "Retry_Limit="`
e.g. `$ ../bin/fluent-bit -i dummy -o es -p "Retry_Limit=" --sosreport`

Without this patch, if we set retry_limit=0, fluent-bit will retry again and again.

Reply:
(By the way, I think this value should be 1 (=default value) at this else case.)

Reply:
> (By the way, I think this value should be 1 (=default value) at this else case.)

I agree

Reply:
Thank you.
I've pushed a commit to set default value 1.